### PR TITLE
Use zypper --xmlout for parsing zypper

### PR DESCRIPTION
**Pull request by fschueller**
Less error prone than dissecting the lines of string output

### CODE REVIEW

File: chef/cookbooks/provisioner/recipes/base.rb [hunk 1]
```diff
@@ -304,15 +304,16 @@
...
-        if attribute == "URI"
-          current_url = value
-        elsif attribute == "Priority"
-          current_priority = value.to_i
-        end
+      out = REXML::Document.new(
+        `LANG=C zypper --non-interactive --xmlout repos Cloud 2> /dev/null`
+      ).root.elements
+
+      current_priority = unless out["repo-list/repo"].attributes["priority"].nil?
```
Comment (MaximilianMeister) [inline chef/cookbooks/provisioner/recipes/base.rb]: I think this check can be dropped, as we can rely on the fact that there is always a url and a priority set for a repository. that would also make hound happy

**Comment by MaximilianMeister** [comment]
+1

File: chef/cookbooks/provisioner/recipes/base.rb [hunk 2]
```diff
@@ -304,16 +304,12 @@
...
-        next if value.nil?
-        if attribute == "URI"
-          current_url = value
-        elsif attribute == "Priority"
-          current_priority = value.to_i
-        end
-      end
+      out = REXML::Document.new(
+        `LANG=C zypper --non-interactive --xmlout repos #{name} 2> /dev/null`
+      ).root.elements
```
Comment (toabctl) [inline chef/cookbooks/provisioner/recipes/base.rb]: What if the repo does not exist? Then I get ```NoMethodError: undefined method `attributes' for nil:NilClass```

Comment (MaximilianMeister) [inline chef/cookbooks/provisioner/recipes/base.rb]: did you get the repo from the provisioner library?

Comment (toabctl) [inline chef/cookbooks/provisioner/recipes/base.rb]: No. Just asking for the case that this happens.

**Comment by nicolasbock** [comment]
+1

**Comment by MaximilianMeister** [comment]
still +1

**Pull request merged**